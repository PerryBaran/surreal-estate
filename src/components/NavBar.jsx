import React from "react";
import { Link } from "react-router-dom";
import style from "../styles/navBar.module.css";
import logo from "../images/logo.png";

const NavBar = () => {
  return (
    <nav className={style.navbar}>
      <img src={logo} alt="logo" className={style.logo} />
      <ul className={style.links}>
        <li className={style.item}>
          <Link to="/">View Properties</Link>
        </li>
        <li className={style.item}>
          <Link to="add-property">Add a Property</Link>
        </li>
      </ul>
    </nav>
  );
};

export default NavBar;
