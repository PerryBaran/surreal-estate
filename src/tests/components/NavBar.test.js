import React from "react";
import { BrowserRouter as Router, matchPath } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import NavBar from "../../components/NavBar";

const RenderWithRouter = (props) => {
  return (
    <Router>
      <NavBar {...props} />
    </Router>
  );
};

describe("NavBar", () => {
  const validProps = {
    handleLogin: jest.fn(),
    handleLogout: jest.fn(),
    userId: "",
  };

  test("snapshot", () => {
    const { asFragment } = render(
      <RenderWithRouter
        handleLogin={validProps.handleLogin}
        handleLogout={validProps.handleLogout}
        userId={validProps.userId}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("links", () => {
    render(
      <RenderWithRouter
        handleLogin={validProps.handleLogin}
        handleLogout={validProps.handleLogout}
        userId={validProps.userId}
      />
    );
    const links = screen.getAllByRole("link");

    expect(screen.getByAltText(/surreal estate/i)).toHaveAttribute(
      "src",
      "logo.png"
    );
    expect(links).toHaveLength(2);
    expect(links[0]).toHaveTextContent(/view properties/i);
    expect(links[0]).toHaveAttribute("href", "/");
    expect(links[1]).toHaveTextContent(/add a property/i);
    expect(links[1]).toHaveAttribute("href", "/add-property");
  });

  test("login rendered when userId is falsey", () => {
    render(
      <RenderWithRouter
        handleLogin={validProps.handleLogin}
        handleLogout={validProps.handleLogout}
        userId={validProps.userId}
      />
    );

    expect(screen.getByText(/login/i)).toBeInTheDocument();
    expect(screen.queryByText(/logout/i)).not.toBeInTheDocument();
  });

  test("logout rendered when userId is truthy", () => {
    validProps.userId = "truthy";
    render(
      <RenderWithRouter
        handleLogin={validProps.handleLogin}
        handleLogout={validProps.handleLogout}
        userId={validProps.userId}
      />
    );

    expect(screen.queryByText(/login/i)).not.toBeInTheDocument();
    expect(screen.getByText(/logout/i)).toBeInTheDocument();
  });
});
