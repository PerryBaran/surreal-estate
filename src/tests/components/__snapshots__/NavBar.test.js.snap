// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <img
      alt="Surreal Estate"
      class="logo"
      src="logo.png"
    />
    <ul
      class="links"
    >
      <li
        class="item"
      >
        <a
          href="/"
        >
          View Properties
        </a>
      </li>
      <li
        class="item"
      >
        <a
          href="/add-property"
        >
          Add a Property
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
